app/models/office.rb

```diff
@@ -1,4 +1,52 @@
 class Office < ActiveRecord::Base
   belongs_to :firm
+
+  validates :email_address,
+    presence: true,
+    length: { maximum: 50 },
+    format: { with: /.+@.+\..+/ }
```

Comment:
:) Yes exactly. It stops subversives like me putting in dan@localhost when testing.